Fix: Docking panels do not reload language

anthonyleestark opened the pull request:
## 🐞Description
The issue was reported in #16780 

## ✅ Test Plan
Manual testing on Windows 10/11

## 📋 Checklist
- [x] Code compiles without errors
- [x] Manual testing completed
- [x] No regressions observed

alankilborn commented:
I noticed when testing the PR code that when changing the localization to Esperanto, the _Search results_ window loses its caption:

![image](https://github.com/user-attachments/assets/9d0d85e2-fd58-43d2-850b-bc6ea6bac859)

This could be because Esperanto doesn't define the text for this (don't know, didn't check that), but, if that's true, shouldn't it remain with English text, not have the text removed?

---

A case where English remains is for the "character panel":

![image](https://github.com/user-attachments/assets/bde44a50-684a-423b-bd9a-1b321d7b4c94)

(Again, didn't check if Esperanto leaves this undefined, or if there is some other problem with the PR code.)

xomx commented on PowerEditor/src/WinControls/DockingWnd/Docking.h:
```diff
@@ -49,7 +50,7 @@
 struct tTbData {
 	HWND hClient = nullptr;                  // client Window Handle
-	const wchar_t* pszName = nullptr;        // name of plugin (shown in window)
+	std::wstring pszName;					 // name of plugin (shown in window)
```
A big NO !
Never try to pass a C++ object ptr thru a C-interface, as it is also intended to be used from other langs than the Notepad++ C++ (N++ plugins use also C#, Pascal, etc...). Your change will lead to a crash of such plugins (STR: install NPPJSONViewer plugin and use its cmd "Show JSON Viewer". Or try with PythonScript and use its "Show Console").

mohannadmir replied:
I m hare and i cant answer and call for transactions . U are a hacker  but i m hare

anthonyleestark commented:
@alankilborn @xomx 

I've updated the code and fixed the bugs you guys pointed out. 
I tested it and it worked for me. But there might be things that I would've missed. 

Can you help me test it again? 
Thank you so much.
